protocol/headers.go

```diff
@@ -42,114 +61,151 @@ type Headers struct {
 
 // CorrelationID returns the 'correlation-id' header value or empty string if not set.
 func (h *Headers) CorrelationID() string {
-	if h.Values[HeaderCorrelationID] == nil {
-		return ""
+	if value, ok := h.Values[HeaderCorrelationID]; ok && value != nil {
+		return value.(string)
 	}
-	return h.Values[HeaderCorrelationID].(string)
+	return ""
 }
 
-// Timeout returns the 'timeout' header value or empty string if not set.
-func (h *Headers) Timeout() string {
-	if h.Values[HeaderTimeout] == nil {
-		return ""
+// Timeout returns the 'timeout' header value or duration of 60 seconds if not set.
+func (h *Headers) Timeout() time.Duration {
+	if value, ok := h.Values[HeaderTimeout]; ok {
+		if duration, err := parseTimeout(value.(string)); err == nil {
+			return duration
+		}
```

Comment:
If we decide to change the API from string to time.Duration - the value in the Headers.Values map must be of type time.Duration as well - not a string one parsed every time. Otherwise, we introduce an inconsistency.
This of course, will requires some marshaling/unmarshaling customizations but it's important to keep the usage of the header the same regardless how the value was accessed - via the getter or via the Headers.Values map directly.